modules/sdk-coin-avaxp/src/lib/delegatorTxBuilder.ts

```diff
@@ -142,36 +145,42 @@ export class DelegatorTxBuilder extends TransactionBuilder {
 
   // endregion
 
-  initBuilder(tx?: AddDelegatorTx): this {
-    if (!tx) return this;
+  initBuilder(tx: Tx): this {
     super.initBuilder(tx);
-    this._nodeID = tx.getNodeIDString();
-    this._startTime = tx.getStartTime();
-    this._endTime = tx.getEndTime();
-    this._stakeAmount = tx.getStakeAmount();
-    this.transaction._utxos = this.recoverUtxos(tx.getIns());
-
+    const baseTx: AddDelegatorTx = tx.getUnsignedTx().getTransaction() as AddDelegatorTx;
+    this._nodeID = baseTx.getNodeIDString();
+    this._startTime = baseTx.getStartTime();
+    this._endTime = baseTx.getEndTime();
+    this._stakeAmount = baseTx.getStakeAmount();
+    this.transaction._utxos = this.recoverUtxos(baseTx.getIns());
     return this;
   }
 
   /**
    *
    * @protected
    */
-  protected buildAvaxpTransaction(): BaseTx {
-    const { inputs, outputs } = this.createInputOutput();
-    return new AddDelegatorTx(
-      this.transaction._networkID,
-      this.transaction._blockchainID,
-      outputs,
-      inputs,
-      this.transaction._memo,
-      utils.NodeIDStringToBuffer(this._nodeID),
-      this._startTime,
-      this._endTime,
-      this._stakeAmount,
-      [this.stakeTransferOut()],
-      this.rewardOwnersOutput()
+  protected buildAvaxpTransaction(): void {
+    const { inputs, outputs, credentials } = this.createInputOutput();
+    this.transaction.setTransaction(
+      new Tx(
```

Comment:
why did you make the change from AddDelegatorTx to new Tx(new Unsigned(new addDelegatorTx))

Reply:
DITTO down